### PR TITLE
calico: Update to Calico 3.17.0

### PR DESCRIPTION
**Pull request by TeddyAndrieux**
Deployment manifest updated based on upstream from
https://docs.projectcalico.org/v3.17/manifests/calico.yaml.

And Calico binaries sha256 updated from
https://github.com/projectcalico/cni-plugin/releases/tag/v3.17.0

---

### CODE REVIEW

File: packages/redhat/calico-cni-plugin.spec [hunk 1]
```diff
@@ -6,12 +6,12 @@
 
 %ifarch x86_64
 %global built_arch              amd64
-%global calico_sha256           baa67cb9fbceb2aebebd6b14e12d448c89dcf694f32b5f450ffd8b780fb3d18d
-%global calico_ipam_sha256      baa67cb9fbceb2aebebd6b14e12d448c89dcf694f32b5f450ffd8b780fb3d18d
+%global calico_sha256           9325d1ead70f2afb9222da64f0c7eaa1f3c5f68763241d6b07cff3a853b783f2
+%global calico_ipam_sha256      9325d1ead70f2afb9222da64f0c7eaa1f3c5f68763241d6b07cff3a853b783f2
```
Comment (NicolasT) [inline packages/redhat/calico-cni-plugin.spec:10]: Same SHA256 for both now? That's surprising...

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 2]
```diff
@@ -3787,6 +3762,10 @@ spec:
         #- name: cni-net-dir
         #  hostPath:
         #    path: /etc/cni/net.d
+        # Used to access CNI logs.
+        - name: cni-log-dir
+          hostPath:
+            path: /var/log/calico/cn
```
Comment (NicolasT) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls]: This seems odd. Not `cni`?

Comment (NicolasT) [reply salt/metalk8s/kubernetes/cni/calico/deployed.sls]: Also, should we create this on the host (using Salt)?

**Review by NicolasT — CHANGES_REQUESTED** [review]

File: packages/redhat/calico-cni-plugin.spec (hunk 1 above)
Comment (TeddyAndrieux) [reply packages/redhat/calico-cni-plugin.spec:10]: Yes and was already the case with 3.16.1

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls (hunk 2 above)
Comment (TeddyAndrieux) [reply salt/metalk8s/kubernetes/cni/calico/deployed.sls]: Ouh right typo on my side, good catch

Hmm I don't know :thinking: I need to test

Comment (TeddyAndrieux) [reply salt/metalk8s/kubernetes/cni/calico/deployed.sls]: We do not need to create the directory it's created automatically

**Review by NicolasT — DISMISSED** [review]

**Review by NicolasT — APPROVED** [review]

**Pull request merged**